src/geometry/GeoTrackView.i.hh

```diff
@@ -157,4 +148,30 @@ CELER_FUNCTION const vecgeom::VPlacedVolume& GeoTrackView::volume() const
 }
 
 //---------------------------------------------------------------------------//
+/*!
+ * Check if current step has crossed a boundary, and update navState if
+ * necessary
+ *
+ * //When using the "cuda"-namespace navigation state (i.e., compiling with
+ * NVCC)
+ * //it's necessary to transform the raw data pointer into an index.
+ */
+CELER_FORCEINLINE_FUNCTION
+bool GeoTrackView::has_same_path()
+{
+    //#### NOT USING YET THE NEW NAVIGATORS ####//
+    // TODO: not using the direction yet here !!
+    bool samepath = vecgeom::GlobalLocator::HasSamePath(
+        detail::to_vector(pos_), vgstate_, vgnext_);
```

Comment:
So I think the term "path" is particlar to VecGeom, meaning the logical state in the hierarchy of volumes, right? It's not clear from the function name or description what is the "same" in `has_same_path` (the current state and the next state?)

It's also confusing to me that this function is written as a boolean query, but it looks like it's really a function to change the state: if you call this function twice in a row, the second call will always return `false` because the first one has updated it?

This function to me looks like it's basically the first part of the `move_next_step()` method that I wrote, and `LinearPropagationHandler::operator()` is the same as `move_next_step()`. So if it makes more sense to split the physical movement (axpy) and logical movement (state CopyTo/UpdateTopMatrix), maybe define `move_next_distance` and `move_next_volume`? Or something similar.

I also don't yet see the use case where `has_same_path` *doesn't* move to the next volume. I think for clarity and performance it would be better to structure the code so it's only called when needed, rather than having checks inside the code. Similarly, I also don't see the use case where the return result (did it move to a new volume?) is needed.

Finally, there's a little bit of funky formatting in the comments above (extra leading slashes), and maybe the `TODO` and `NOT USING YET` could be elaborated on or removed as appropriate.

Reply:
> Similarly, I also don't see the use case where the return result (did it move to a new volume?) is needed.

In GeantV's FieldPropagationHandler::DoIt, the result is used because:
```
    // Crossing tracks continue to continuous processes, the rest have to
    // query again the geometry 
```
I.e., in general, within a step that flow is completely different whether you cross a volume on or not.  [However, that is not saying that the information is not already (could be) there from other source]

Reply:
The term path is used in VecGeom.  I agree that _volume_ or _logical volume_ makes more sense.
The use case for staying in the same path is for charged tracks, for which moving by 'proposed step' changes due to track curvature. The proposed step is broken up into smaller linear substeps along the curve, and the track may change volumes along the way, so the actual trajectory to boundary can be both larger or smaller than proposed step, so it needs to be tested.
Your suggestion to break 'move_next_step()' into 'move_next_distance()' and 'move_next_volume()' makes sense, so I will work on that.